### ISSUE TITLE
exported: `main` package

### ISSUE THREAD

**Issue by ccoVeille** [mgechev/revive#1397]
symbols declared in `main` package [cannot be imported](https://go.dev/ref/spec#Program_execution).

It's the same for the content of `_test` files.

Here is a repository

https://github.com/CodeWithEmad/gh-fork-cleanup/blob/v1.1.0/main.go

It contains exported structures.

Right now, revive 1.10.0 reports the following errors

```
main.go:16:6: exported: exported type Owner should have comment or be unexported (revive)
type Owner struct {
     ^
main.go:21:6: exported: exported type Repo should have comment or be unexported (revive)
type Repo struct {
```

Test files are already excluded in `exported` rule
https://github.com/mgechev/revive/blob/v1.10.0/rule/exported.go#L114

There are multiple ways to consider this:
- `exported` should ignore a file when a file in `main`
- `exported` should report that these structures are exported in a package that cannot be exported.
- other ideas we may think about.

**Comment by ccoVeille** [comment]
The exact issue occurs with `unexported-return` rule

```
package main

// foo is a stub
type foo struct{}

// NewFoo is a stub
func NewFoo() foo { // unexported-return: exported func NewFoo returns unexported type main.foo, which can be annoying to use
	return foo{}
}
```
